qemu/tests/migration_with_block.py

```diff
@@ -0,0 +1,233 @@
+        """ Get the data disks in windows. """
+        for data_image in params['images'].split()[1:]:
+            size = params.object_params(data_image)['image_size']
+            yield utils_disk.get_windows_disks_index(session, size)[0], size
+
+    def get_data_disks():
+        """ Get the data disks. """
+        _get_disks = _get_data_disks_win if windows else _get_data_disks_linux
+        for disk, size in _get_disks():
+            yield disk, size
+
+    def format_data_disks():
+        """ Format the data disks. """
+        for disk, size in get_data_disks():
+            if windows:
+                if not utils_disk.update_windows_disk_attributes(session, disk):
+                    test.fail("Failed to update windows disk attributes.")
+            yield utils_disk.configure_empty_disk(session, disk, size, os_type)[0]
+
+    def iozone_test():
+        """ Do iozone testing inside guest. """
+        error_context.context("Do iozone testing on data disks.", logging.info)
+        iozone = generate_instance(params, vm, 'iozone')
+        try:
+            for target in format_data_disks():
+                iozone.run(params['iozone_options'].format(target))
+        finally:
+            iozone.clean()
+
+    def get_cdrom_size():
+        """ Get the size of cdrom device inside guest. """
+        error_context.context("Get the cdrom's size in guest.", logging.info)
+        cmd = params["check_size"]
+        if not utils_misc.wait_for(
+                lambda: re.search(r'(\d+)', session.cmd(cmd), re.M), 10):
+            test.fail('Failed to get the cdrom\'s size.')
+        cdrom_size = re.search(r'(\d+)', session.cmd(cmd), re.M).group(1)
+        cdrom_size = int(cdrom_size) * 512 if not windows else int(cdrom_size)
+        logging.info("The cdrom's size is %s in guest.", cdrom_size)
+        return cdrom_size
+
+    def get_iso_size(iso_file):
+        """ Get the size of iso on host."""
+        error_context.context("Get the iso size on host.", logging.info)
+        return int(process.system_output(
+            'ls -l %s | awk \'{print $5}\'' % iso_file, shell=True).decode())
+
+    def compare_cdrom_size(iso_file):
+        """ Compare the cdrom's size between host and guest. """
+        error_context.context(
+            "Compare the cdrom's size between host and guest.", logging.info)
+        ios_size = get_iso_size(iso_file)
+        if not utils_misc.wait_for(lambda: get_cdrom_size() == ios_size, 30, step=3):
+            test.fail('The size inside guest is not equal to iso size on host.')
+        return get_cdrom_size()
+
+    def check_cdrom_info_by_qmp(check_items):
+        """ Check the cdrom device info by qmp. """
+        error_context.context(
+            'Check if the info \"%s\" are match with the output of query-block.' %
+            str(check_items), logging.info)
+        blocks = vm.monitor.info_block()
+        for key, val in check_items.items():
+            if key == 'device':
```

Comment:
```python
if key != 'device or val != dev_id or blocks[dev_id][key] != val:
    test.fail()
```
should be same as yours

Reply:
if key == 'device' and val != dev_id or blocks[dev_id][key] != val:
    test.fail()

may be like this? because 
if key != 'device' and blocks[dev_id][key] == val:
    continue